userspace/libsinsp/sinsp.cpp

```diff
@@ -1864,10 +1864,12 @@ uint32_t sinsp::reserve_thread_memory(uint32_t size)
 
 void sinsp::get_capture_stats(scap_stats* stats) const
 {
-	if(scap_get_stats(m_h, stats) != SCAP_SUCCESS)
-	{
-		throw sinsp_exception(scap_getlasterr(m_h));
-	}
+	scap_get_stats(m_h, stats);
```

Comment:
Why do we no longer throw an exception here and silently ignore failures?

Reply:
Same here, probably we can go on even if we are not able to collect stats, since this is well tested we could keep also the old approach here :man_shrugging:

Reply:
Ah, that's a good point. I guess if we can't get the stats, we can consider it a successful report of zero stats instead. Let's just make sure we get *nstats right (in scap_stats_v2).

Still, we're still ignoring the stats failures here,  just in a different place. We can keep this as is but please add a comment that we're explicitly ignoring errors and why. Ideally, we'd throw the exception and let the caller handle it but we can leave it for a future PR (that will subtly break the sinsp API so let's make that one small and explicit).

Reply:
Agree here, we need to choose a policy to handle errors and apply it to both approaches, in the meanwhile I would revert this as it was